### ISSUE TITLE
WhereIn Typings

### ISSUE THREAD

**Issue by danieloprado** [Vincit/objection.js#456]
Before update, the query above is correctly, but now **whereIn** expect a query based on Event this is not right because I can do a subquery in another table.

![image](https://user-images.githubusercontent.com/7683575/28668539-b09625d2-72a6-11e7-8ea0-145a60e3b5f3.png)

**Comment by danieloprado** [comment]
Typescript error:

```
Argument of type 'QueryBuilder<EventDate>' is not assignable to parameter of type 'QueryBuilder<Event>'.
  Type 'EventDate' is not assignable to type 'Event'.
    Property 'id' is missing in type 'EventDate'.
```

**Comment by koskimas** [comment]
Fixed in 5e33cbc
